File: src/test/java/calculator2/StringCalculator.java
package calculator2;


import java.util.stream.Stream;

public class StringCalculator {

    private final TextInputHandler textInputHandler = new TextInputHandler();
    private final NumberMapper numberMapper = new NumberMapper();
    private final Calculator calculator = new Calculator();

    public int add(String text) {
        if (textInputHandler.isBlank(text)) {
            return 0;
        }
        String[] tokens = textInputHandler.tokenize(text);
        Stream<Number> numbers = numberMapper.toNumbers(tokens);
        return calculator.sum(numbers);
    }
}

File: src/main/java/calculator/StringCalculator.kt
package calculator

import calculator.extractor.TextDelimiterExtractorFactory

class StringCalculator(
    private val factory: TextDelimiterExtractorFactory,
) {

    fun add(text: String?): Int {
        return factory.get(text)
            .extract(text ?: "")
            .sum().value
    }
}
